Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44859&attempt_id=36daf3b0-c51e-4187-aaac-4512bcd8e42e)

Attempted: mosml

<details><summary>Partial log (click to expand)</summary><p>

```
make[1]: Entering directory '/private/tmp/nix-build-mosml-2.10.1.drv-0/mosml-ver-2.10.1/src/config'
sh autoconf gcc
Checking the sizes of integers and pointers...
This architecture seems to be neither 32 bits nor 64 bits.
Caml Light won't run on this architecture.
make[1]: *** [Makefile:13: runtime] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-mosml-2.10.1.drv-0/mosml-ver-2.10.1/src/config'
make: *** [Makefile:26: world] Error 2
builder for '/nix/store/xi9md20vq8fsy93da1rg74nqvy4xnc1y-mosml-2.10.1.drv' failed with exit code 2
error: build of '/nix/store/xi9md20vq8fsy93da1rg74nqvy4xnc1y-mosml-2.10.1.drv' failed
```
</p></details>

--- a/pkgs/development/compilers/mosml/default.nix
+++ b/pkgs/development/compilers/mosml/default.nix
@@ -1,0 +1,29 @@
+{ stdenv, fetchurl, gmp, perl }:
+
+stdenv.mkDerivation rec {
+  name = "mosml-${version}";
+  version = "2.10.1";
+
+  buildInputs = [ gmp perl ];
+
+  makeFlags = "PREFIX=$(out)";
+
+  src = fetchurl {
+    url = "https://github.com/kfl/mosml/archive/ver-${version}.tar.gz";
+    sha256 = "13x7wj94p0inn84pzpj52dch5s9lznqrj287bd3nk3dqd0v3kmgy";
+  };
+
+  setSourceRoot = ''export sourceRoot="$(echo */src)"'';
+
+  meta = with stdenv.lib; {
+    description = "A light-weight implementation of Standard ML";
+    longDescription = ''
+      Moscow ML is a light-weight implementation of Standard ML (SML), a strict
+      functional language used in teaching and research.
+    '';
+    homepage = http://mosml.org/;
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ vaibhavsagar ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7014,6 +7014,8 @@ with pkgs;
 
   monoDLLFixer = callPackage ../build-support/mono-dll-fixer { };
 
+  mosml = callPackage ../development/compilers/mosml { };
+
   mozart-binary = callPackage ../development/compilers/mozart/binary.nix { };
   mozart = mozart-binary;
 

